src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -126,32 +132,107 @@ public void initInPlace(Map<K, V> reference, PseudoRandom prng) {
       }
     }
 
+    private void eraseRandomChunk(Map<K, V> map, PseudoRandom prng, int mapSize) {
+      int upperLimit = Math.max(1, Math.min(mapSize - minSize, mapSize / 2));
+      int chunkSize = upperLimit == 1 ? 1 : prng.closedRange(1, upperLimit);
+      int chunkOffset = prng.closedRange(0, mapSize - chunkSize);
+
+      List<K> tmpRemStore = map.entrySet()
+                                .stream()
+                                .skip(chunkOffset)
+                                .limit(chunkSize)
+                                .map(Map.Entry::getKey)
+                                .collect(Collectors.toCollection(() -> new ArrayList<>(chunkSize)));
+
+      map.keySet().removeAll(tmpRemStore);
+    }
+
+    private void insertRandomChunk(Map<K, V> map, PseudoRandom prng, int mapSize) {
+      int chunkSize = prng.closedRange(1, Math.min(maxSize - mapSize, mapSize));
+      for (int i = 0; i < chunkSize; i++) {
+        initElement(map, prng);
+      }
+    }
+
+    private void changeRandomChunk(Map<K, V> map, PseudoRandom prng, int mapSize) {
+      int chunkOffset = prng.indexIn(mapSize);
+      int chunkSize =
+          Math.min(prng.closedRange(1, mapSize - chunkOffset), (int) Math.ceil(mapSize / 10.0));
+
+      List<K> keys = new ArrayList<>(map.keySet());
+      int elementsChanged = 0;
+
+      for (int i = chunkOffset; i < chunkOffset + chunkSize && i < keys.size(); i++) {
```

Comment:
Is the comparison with `keys.size()` necessary? `chunkOffset + chunkSize` should be at most the initial map size and `keys` doesn't change in this loop.